src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -262,6 +371,51 @@ void ProtocolHandlerImpl::SendEndSessionNAck(ConnectionID connection_id,
                     << static_cast<int32_t>(session_id));
 }
 
+void ProtocolHandlerImpl::SendEndSessionNAck(
+    ConnectionID connection_id,
+    uint32_t session_id,
+    uint8_t protocol_version,
+    uint8_t service_type,
+    std::vector<std::string>& rejectedParams) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ProtocolFramePtr ptr(
+      new protocol_handler::ProtocolPacket(connection_id,
+                                           protocol_version,
+                                           PROTECTION_OFF,
+                                           FRAME_TYPE_CONTROL,
+                                           service_type,
+                                           FRAME_DATA_END_SERVICE_NACK,
+                                           session_id,
+                                           0u,
+                                           message_counters_[session_id]++));
+  if (rejectedParams.size() > 0) {
```

Comment:
Does there need to be a check for protocol version >= 5 like in StartSessionAck()?